test/desdemona/query_test.clj

```diff
@@ -29,31 +29,93 @@
+    #{}
+    '()
+
+    #{'x}
+    (#'q/dsl->logic '(= (:ip x) "10.0.0.1"))
+
+    #{'y}
+    (#'q/dsl->logic '(= (:ip y) "10.0.0.1"))
+
+    #{'x}
+    (#'q/dsl->logic '(= "10.0.0.1" (:ip x)))
+
+    #{'x}
+    (#'q/dsl->logic '(= (:type x) "egress"))
+
+    #{'x}
+    (#'q/dsl->logic '(and (= (:ip x) "10.0.0.1")
+                          (= (:type x) "egress")))
+
+    #{'x 'y}
+    (#'q/dsl->logic '(= (:ip x) (:ip y)))))
+
+(defmacro with-fake-gensym
+  [& body]
+  `(let [gensym-count# (atom 0)
+         fake-gensym# (fn [& args#]
+                        (->> (swap! gensym-count# inc)
+                             (str "fake-gensym-")
+                             (symbol)))]
+     (with-redefs [clojure.core/gensym fake-gensym#]
+       ~@body)))
 
 (deftest dsl->logic-tests
-  (is (thrown? IllegalArgumentException (dsl->logic '(BOGUS BOGUS BOGUS))))
+  (is (thrown? IllegalArgumentException
+               (#'q/dsl->logic '(BOGUS BOGUS BOGUS))))
   (is (= '(clojure.core.logic/featurec x {:ip "10.0.0.1"})
-         (dsl->logic '(= (:ip x) "10.0.0.1"))
-         (dsl->logic '(= "10.0.0.1" (:ip x)))))
+         (#'q/dsl->logic '(= (:ip x) "10.0.0.1"))
+         (#'q/dsl->logic '(= "10.0.0.1" (:ip x)))))
+  (testing "logic variable is not hard coded to 'x"
+    (is (= '(clojure.core.logic/featurec y {:ip "10.0.0.1"})
+           (#'q/dsl->logic '(= (:ip y) "10.0.0.1"))
+           (#'q/dsl->logic '(= "10.0.0.1" (:ip y))))))
   (testing "logical conjunction"
     (is (= '(clojure.core.logic/conde
              [(clojure.core.logic/featurec x {:ip "10.0.0.1"})
               (clojure.core.logic/featurec x {:type "egress"})])
-           (dsl->logic '(and (= (:ip x) "10.0.0.1")
-                             (= (:type x) "egress"))))))
+           (#'q/dsl->logic '(and (= (:ip x) "10.0.0.1")
+                                 (= (:type x) "egress"))))))
   (testing "logical disjunction"
     (is (= '(clojure.core.logic/conde
              [(clojure.core.logic/featurec x {:ip "10.0.0.1"})]
              [(clojure.core.logic/featurec x {:type "egress"})])
-           (dsl->logic '(or (= (:ip x) "10.0.0.1")
-                            (= (:type x) "egress")))))
+           (#'q/dsl->logic '(or (= (:ip x) "10.0.0.1")
+                                (= (:type x) "egress")))))
     (is (= '(clojure.core.logic/conde
              [(clojure.core.logic/featurec x {:type "egress"})]
              [(clojure.core.logic/featurec x {:ip "10.0.0.1"})])
-           (dsl->logic '(or (= (:type x) "egress")
-                            (= (:ip x) "10.0.0.1")))))))
+           (#'q/dsl->logic '(or (= (:type x) "egress")
+                                (= (:ip x) "10.0.0.1"))))))
+  (testing "multiple literals"
+    (is (thrown? IllegalArgumentException
+                 (#'q/dsl->logic '(= (:ip x) "10.0.0.1" "10.0.0.1"))))
+    (is (thrown? IllegalArgumentException
+                 (#'q/dsl->logic '(= (:ip x) "10.0.0.1" "10.0.0.2")))))
```

Comment:
Ah - I thought I scanned it more thoroughly - I didn't realize the ips only differ by the trailing `1`.